kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/parquet/ParquetRecordWriterProvider.java

```diff
@@ -57,19 +67,29 @@ public String getExtension() {
   @Override
   public RecordWriter getRecordWriter(final S3SinkConnectorConfig conf, final String filename) {
     return new RecordWriter() {
+      final String adjustedFilename = getAdjustedFilename(filename, getExtension());
       Schema schema = null;
       ParquetWriter<GenericRecord> writer;
       S3ParquetOutputFile s3ParquetOutputFile;
 
       @Override
       public void write(SinkRecord record) {
         if (schema == null) {
-          schema = record.valueSchema();
+          schema = recordView.getViewSchema(record);
+          // pad the schema for AvroParquetWriter that only accepts record types
+          if (recordView instanceof HeaderRecordView) {
+            schema = SchemaBuilder.struct().name(HEADER_STRUCT_NAME)
+                .field(HEADER_FIELD_NAME, schema).build();
+          } else if (recordView instanceof KeyRecordView) {
+            schema = SchemaBuilder.struct().name(KEY_STRUCT_NAME)
```

Comment:
how do we know that they key at least is not of record type? If someone chooses parquet for keys, we'd assume they have they keys in the appropriate type (like the values).

Reply:
updated code with a check for the type